State restructure

Exulansis opened the pull request:
Improvements suggested in #33.

1. The `device` state entry was removed in favor of filtering `devices`
2. The `devices` entry was changed from `[device, device, device]` to 
```
{
  deviceId: {device},
  deviceId: {device},
  ...
}
```
This allows us to do some operations in constant time (i.e `this.setState({device: devices[this.props.params.deviceId]})` instead of having to use `devices.find(d => d.deviceId === this.props.params.deviceId)`).
This also allows us to updated an individual device entry easier.

This is an initial version, would be really happy to get feedback, and can easily add modifications if required..

Exulansis committed:
State restructured to remove unnecessary elements and reduce complexity.

connects to #33

Change-type: patch

craigmulligan reviewed (approved):
Nice 👍

craigmulligan commented:
Oh @Exulansis, can you make sure to add a versionist compatible commit message, probably best to squash and amend with a something like:
```
refactor redux state structure

connects to #33

Change-type: minor
```

This way the ci, will automatically deploy a new version on dockerhub etc.

Commit:
fix device-type hammering

connects to #36
Also updated travis scripts so #32 be fixed too

Change-Type: patch
